### ISSUE TITLE
NIX_PATH not honored

### ISSUE THREAD

**Issue by zhaofengli** [nix-community/nix-eval-jobs#324]
On 8802412b8747633e9d80639897e4d58fa6290909, `NIX_PATH` does not appear to have an effect:

```console
$ NIX_PATH=meow=/dev/null ./build/src/nix-eval-jobs -E 'toString <meow>'
...
              error: file 'meow' was not found in the Nix search path (add it using $NIX_PATH or -I)
```

**Comment by Mic92** [comment]
This is a "feature" that inherited from the original hydra-eval-jobs: https://github.com/nix-community/nix-eval-jobs/blob/8802412b8747633e9d80639897e4d58fa6290909/src/nix-eval-jobs.cc#L348

**Comment by zhaofengli** [comment]
Oh wow, my grep-foo failed and it worked before. CI relying on `<nixpkgs>` only started to fail in Colmena since the last bump.

I guess `NIX_PATH` previously working should be considered unintentional despite the error massage?

**Comment by Mic92** [comment]
Maybe NIX_PATH was interpreted before that line of code?
What should still work is `nix-eval-jobs -I nixpkgs=` etc?
Not sure what the best course of action here is. I would like to use nix-eval-jobs in hydra again at some point.

**Comment by zhaofengli** [comment]
Yeah, it was interpreted when constructing the global `evalSettings`. IMO nix-eval-jobs should act as "normal Nix" and accept the variable and let the user (e.g., Hydra) clear it as needed. The user can opt to use flakes/pure-eval/restricte-eval if they care about hermeticity.

**Comment by Mic92** [comment]
Feel free to make a pull request removing unsetenv and updating a documentation on how to get hermetic evaluation.
